src/ui/dialogs/exstatus.ts

```diff
@@ -0,0 +1,68 @@
+import Dialog from '../Dialog'
+import Client from '../../Client'
+//import JID from '../../JID'
+
+let exstatusTemplate = require('../../../template/exstatus.hbs');
+
+let dialog: Dialog;
+
+export default function() {
+
+   let content = exstatusTemplate();
+
+   dialog = new Dialog(content);
+   let dom = dialog.open();
+
+   $(dom).ready(function(){
+    getStatus();
+   });
+
+   dom.find('form').submit(onSubmit);
+
+   $('.jsxc-js-clear').on('click',removeStatus);
+}
+
+function onSubmit()
+{
+   let account = Client.getAccountManager().getAccounts()[0];
+   let statustext = dialog.getDom().find('#extended_status_text').val().toString();
+   let item = $build('status',{xmlns:'http://jabber.org/protocol/status'}).c('text').t(statustext).tree();
```

Comment:
We should move this to a service.

Reply:
ok i have moved the build code for the packets to pep service